pkg/webui/lib/components/with-auth.js

```diff
@@ -47,14 +45,16 @@ class Auth extends React.PureComponent {
     }
 
     if (!Boolean(user)) {
-      return (
-        <Redirect
-          to={{
-            pathname: `${env.app_root}/login`,
-            state: { from: this.props.location.pathname },
-          }}
-        />
-      )
+      // Compose the redirect path, to send the user back to after login
+      const redirectPath = window.location.pathname.substring(env.app_root.length)
+      let loc = `${env.app_root}/login`
+      if (redirectPath !== '') {
+        loc += `?next=${redirectPath}`
+      }
+
+      // We need a full reload here to ensure the new state cookie is being set
+      window.location = loc
```

Comment:
this causes nasty flashing with empty breadcrumbs being visible

Reply:
Yup. We actually don't need the hard redirect anymore, as the auth query is not passed as `PAGE_DATA` anymore. Using react router redirect solves the issue of flashing.